postgresql/resource_postgresql_script.go

```diff
@@ -103,10 +121,12 @@ func executeCommands(db *DBConnection, commands []string, tries int, backoffDela
 	}
 }
 
-func executeBatch(db *DBConnection, commands []string) error {
+func executeBatch(ctx context.Context, db *DBConnection, commands []string, timeout int) error {
+	timeoutContext, timeoutCancel := context.WithTimeout(ctx, time.Duration(timeout)*time.Second)
+	defer timeoutCancel()
```

Comment:
Should you issue rollback in term of cancellation?

Reply:
We I see the test, seems to be okay.

Reply:
since the context only cancel the `ExecuteWithContext`, it returns with an error and is catch later.
the `defer` only declare a function to call at the end of the function.